plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/widget/StringArrayWidget.java

```diff
@@ -0,0 +1,29 @@
+package edu.wpi.first.shuffleboard.plugin.base.widget;
+
+import edu.wpi.first.shuffleboard.api.widget.Description;
+import edu.wpi.first.shuffleboard.api.widget.SimpleAnnotatedWidget;
+import edu.wpi.first.shuffleboard.plugin.base.control.ArrayTableView;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.StackPane;
+
+@Description(
+    name = "String Array",
+    dataTypes = String[].class,
+    summary = "Displays an array of strings"
+)
+public final class StringArrayWidget extends SimpleAnnotatedWidget<String[]> {
+  private final StackPane pane = new StackPane();
+  private final ArrayTableView<String> table = new ArrayTableView<>();
+
+  @SuppressWarnings("JavadocMethod")
+  public StringArrayWidget() {
+    pane.getChildren().add(table);
+
+    dataOrDefault.addListener((observableValue, oldStrings, newStrings) -> table.setItems(newStrings));
```

Comment:
Why do the boolean and number array widgets make copies of the input values, but not the string widget?

Reply:
https://github.com/Martysh12/shuffleboard/blob/6599d321b23ea3fe2679fcf508ad2bb8af4ae727/plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/widget/BooleanArrayWidget.java#L40-L44
```java
    dataOrDefault.addListener((observableValue, oldBooleans, newBooleans) -> {
      final var array = new Boolean[newBooleans.length];
      IntStream.range(0, newBooleans.length).forEach(i -> array[i] = newBooleans[i]);
      table.setItems(array);
    });
```

I'm forced to make a copy for the boolean widget, because `newBooleans` is a `boolean[]`, while `setItems` accepts a `Boolean[]`, because:
1. I'm unable to put a primitive as a type argument here, which would change `setItems`'s argument:
https://github.com/Martysh12/shuffleboard/blob/6599d321b23ea3fe2679fcf508ad2bb8af4ae727/plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/widget/BooleanArrayWidget.java#L27
```java
  private final ArrayTableView<Boolean> table = new ArrayTableView<>();
```
2. I'm unable to put `Boolean[]` as the type argument for `SimpleAnnotatedWidget` (which would change the type of `newBooleans`) because it throws an exception:
https://github.com/Martysh12/shuffleboard/blob/6599d321b23ea3fe2679fcf508ad2bb8af4ae727/plugins/base/src/main/java/edu/wpi/first/shuffleboard/plugin/base/widget/BooleanArrayWidget.java#L25
```java
public final class BooleanArrayWidget extends SimpleAnnotatedWidget<boolean[]> {
```
```
WARNING: Uncaught exception on JavaFX Application Thread
java.lang.ClassCastException: class [Z cannot be cast to class [Ljava.lang.Boolean; ([Z and [Ljava.lang.Boolean; are in module java.base of loader 'bootstrap')
	at com.sun.javafx.binding.ExpressionHelper$SingleChange.fireValueChangedEvent(ExpressionHelper.java:181)
	at com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(ExpressionHelper.java:80)
	at javafx.beans.binding.ObjectBinding.invalidate(ObjectBinding.java:176)
	at com.sun.javafx.binding.BindingHelperObserver.invalidated(BindingHelperObserver.java:52)
	at com.sun.javafx.binding.ExpressionHelper$SingleInvalidation.fireValueChangedEvent(ExpressionHelper.java:136)
	at com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(ExpressionHelper.java:80)
	at javafx.beans.binding.ObjectBinding.invalidate(ObjectBinding.java:176)
	at org.fxmisc.easybind.monadic.FlatMapBindingBase.mappedInvalidated(FlatMap.java:76)
	at org.fxmisc.easybind.monadic.FlatMapBindingBase.lambda$new$22(FlatMap.java:21)
	at javafx.beans.WeakInvalidationListener.invalidated(WeakInvalidationListener.java:83)
	at edu.wpi.first.shuffleboard.api.properties.AtomicPropertyListenerDelegate.lambda$invalidated$1(AtomicPropertyListenerDelegate.java:76)
	at edu.wpi.first.shuffleboard.api.util.FxUtils.runOnFxThread(FxUtils.java:60)
	at edu.wpi.first.shuffleboard.api.util.AsyncUtils.runAsync(AsyncUtils.java:24)
	at edu.wpi.first.shuffleboard.api.properties.AtomicPropertyListenerDelegate.lambda$invalidated$2(AtomicPropertyListenerDelegate.java:76)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at edu.wpi.first.shuffleboard.api.properties.AtomicPropertyListenerDelegate.invalidated(AtomicPropertyListenerDelegate.java:76)
	at edu.wpi.first.shuffleboard.api.properties.AsyncProperty.set(AsyncProperty.java:77)
	at edu.wpi.first.shuffleboard.api.properties.AsyncProperty.setValue(AsyncProperty.java:82)
	at edu.wpi.first.shuffleboard.api.sources.DataSource.setData(DataSource.java:101)
	at edu.wpi.first.shuffleboard.plugin.networktables.sources.SingleKeyNetworkTableSource.lambda$new$0(SingleKeyNetworkTableSource.java:50)
	at edu.wpi.first.shuffleboard.plugin.networktables.sources.NetworkTableSource.lambda$setTableListener$0(NetworkTableSource.java:87)
	at edu.wpi.first.shuffleboard.api.util.FxUtils.lambda$runOnFxThread$0(FxUtils.java:64)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:457)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:456)
	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(GtkApplication.java:290)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

Reply:
Ahh, right, the Java type system bites us here. This is fine, then